### ISSUE TITLE
Small Parquet file write times increased from under 10 seconds to over 200 seconds 

### ISSUE THREAD

**Issue by davidbw246** [noobaa/noobaa-core#6690]
### Environment info
- noobaa version
INFO[0000] CLI version: 5.9.0                           
INFO[0000] noobaa-image: noobaa/noobaa-core:master-20210726 
INFO[0000] operator-image: noobaa/noobaa-operator:5.9.0 

- Platform: OCP 4.7.4

### Actual behavior
1. PUT of small parquet files(<100KB) always takes over 200 seconds.

### Expected behavior
1. PUT of small parquet files(<100KB) take under 10 seconds.

### Steps to reproduce
Install noobaa
Run with Spark + Hadoop (currently spark-3.0.0-bin-without-hadoop + hadoop-3.1.2) - hadoop is only used to provide the s3a shim for Spark.
Create a dataframe with 100KB of data
dataframe.write.parquet()

### More information - Screenshots / Logs / Other output
Writes for files that always took a few seconds on noobaa version 0609 always take over 200 seconds on noobaa 0726. The endpoint log shows it does a PUT OBJECT, then after 200 seconds does a second PUT OBJECT for the same object. Maybe there is a 200 second timeout then it re-retries?

Here is an example from the endpoint log for spark writing a small(<2MB) parquet file:
^[[32mJul-29 14:40:09.141^[[35m [Endpoint/9] ^[[36m   [L0]^[[39m core.endpoint.s3.ops.s3_put_object:: PUT OBJECT s3bucket catalog_page.parquet/_temporary/0/

^[[32mJul-29 14:40:09.768^[[35m [Endpoint/9] ^[[36m   [L0]^[[39m core.endpoint.s3.ops.s3_put_object:: PUT OBJECT s3bucket catalog_page.parquet/_temporary/0/_temporary/attempt_20210729074009_0004_m_000000_10/part-00000-1b661bb4-7217-4c63-b945-2bb6bfefdeb1-c000.snappy.parquet

^[[32mJul-29 14:40:10.029^[[35m [Endpoint/9] ^[[36m   [L0]^[[39m core.endpoint.s3.ops.s3_put_object:: **PUT OBJECT s3bucket catalog_page.parquet/part-00000-1b661bb4-7217-4c63-b945-2bb6bfefdeb1-c000.snappy.parquet /s3bucket/catalog_page.parquet/_temporary/0/_temporary/attempt_20210729074009_0004_m_000000_10/part-00000-1b661bb4-7217-4c63-b945-2bb6bfefdeb1-c000.snappy.parquet**

200 second wait:

^[[32mJul-29 14:43:30.184^[[35m [Endpoint/9] ^[[36m   [L0]^[[39m core.endpoint.s3.ops.s3_put_object:: **PUT OBJECT s3bucket catalog_page.parquet/part-00000-1b661bb4-7217-4c63-b945-2bb6bfefdeb1-c000.snappy.parquet /s3bucket/catalog_page.parquet/_temporary/0/_temporary/attempt_20210729074009_0004_m_000000_10/part-00000-1b661bb4-7217-4c63-b945-2bb6bfefdeb1-c000.snappy.parquet**

^[[32mJul-29 14:43:30.271^[[35m [Endpoint/9] ^[[36m   [L0]^[[39m core.endpoint.s3.ops.s3_put_object:: PUT OBJECT s3bucket catalog_page.parquet/_temporary/0/_temporary/attempt_20210729074009_0004_m_000000_10/
^[[32mJul-29 14:43:30.441^[[35m [Endpoint/9] ^[[36m   [L0]^[[39m core.endpoint.s3.ops.s3_put_object:: PUT OBJECT s3bucket catalog_page.parquet/_SUCCESS

**Comment by davidbw246** [comment]
My attempts to attached "oc adm must-gather" output to this issue fail with "is not included in the list". Maybe the file is too big(it is 127MB)? I can email it on request.

**Comment by davidbw246** [comment]
[must-gather-Aug3.tar.gz](https://github.com/noobaa/noobaa-core/files/6923797/must-gather-Aug3.tar.gz)

I reran mustgather with --image=registry.redhat.io/ocs4/ocs-must-gather-rhel8 and the file was small enough to attach.

**Comment by jackyalbo** [comment]
@davidbw246 So it seems that currently you system has issues: The DB pod is down. And the logs of the core pod consists only failures on trying to access the DB
From the endpoint logs I can see issues with permission denied on accessing the bucket1 sub-directories. like here:
 ```
Jul-29 13:51:42.453 [Endpoint/9] [ERROR] core.sdk.namespace_fs:: check_access: error  EACCES [Error: Permission denied] { code: 'EACCES' } /nsfs/fs1/bucket1/call_center.parquet/ /nsfs/fs1/bucket1
```
Did you find any of the files really uploaded to the storage?
Can you start the same test and before starting spark load just try to upload a single object to a bucket to see everything is setup correctly. 
Also if you can share all the logs inside the core pod under /log - it consists of older logs that may have information on the time you created bucket1 - and why we don't have access to it.

**Comment by davidbw246** [comment]
@jackyalbo /log/noobaa.log on the noobaa core pod was too big to attach to this issue so I emailed it to you at Jacky Albo/Israel/Contr/IBM@IBM.

The DB pod was up when I ran the Spark workloads last week and again today and I still see this issue. The storage I am using must have been down when I ran must gather, which would have caused the DB pod to go down. Should I rerun must gather?

Regarding the error you mention above for "call_center.parquet": Creating that parquet file requires reading object "call_center.dat", which has size 1891 bytes. I am seeing noobaa issue 6680(InvalidObjectState) when I run with objects smaller than 4KB, which causes all of my tests with .dat files less than 4KB to fail and is unrelated to this performance issue.

I am able to "aws s3 cp" objects larger than 4KB to and from this cluster and I can logon to the endpoint pod and see that the files exist in /nsfs/fs1/bucket1:
          Command below run from my endpoint pod:
          sh-4.4# ls -l /nsfs/fs1/bucket1/call_center.dat
          -rw-r-----. 1 2001 2002 1891 Jul 29 18:11 /nsfs/fs1/bucket1/call_center.dat

I can also see the .parquet files I created in that directory. 
The problem I am reporting in this issue is the 200 second delay before it creates the files.

**Comment by jackyalbo** [comment]
@davidbw246 Sadly I don't really have access to that email. Please send it to jalbo@redhat.com

**Comment by davidbw246** [comment]
@jackyalbo ok - sent to  jalbo@redhat.com just now

**Comment by jackyalbo** [comment]
```
Jul-29 14:40:10.029 [Endpoint/9]    [L0] core.endpoint.s3.ops.s3_put_object:: PUT OBJECT s3bucket catalog_page.parquet/part-00000-1b661bb4-7217-4c63-b945-2bb6bfefdeb1-c000.snappy.parquet /s3bucket/catalog_page.parquet/_temporary/0/_temp
orary/attempt_20210729074009_0004_m_000000_10/part-00000-1b661bb4-7217-4c63-b945-2bb6bfefdeb1-c000.snappy.parquet
Jul-29 14:40:10.034 [Endpoint/9]    [L0] core.sdk.namespace_fs:: _fail_if_archived_or_sparse_file: /nsfs/fs1/bucket1/catalog_page.parquet/_temporary/0/_temporary/attempt_20210729074009_0004_m_000000_10/part-00000-1b661bb4-7217-4c63-b945-2bb6bfefdeb1-c000.snappy.parquet rejected { dev: 4194559, ino: 219192, mode: 33184, nlink: 1, uid: 2001, gid: 2002, rdev: 0, size: 695113, blksize: 4194304, blocks: 0, atimeMs: 1627569609858.087, ctimeMs: 1627569609868.325, mtimeMs: 16
27569609866.732, birthtimeMs: 1627569609868.325, atime: 2021-07-29T14:40:09.858Z, mtime: 2021-07-29T14:40:09.867Z, ctime: 2021-07-29T14:40:09.868Z, birthtime: 2021-07-29T14:40:09.868Z }
(node:9) UnhandledPromiseRejectionWarning: Error: Attempted to access archived or sparse file
    at NamespaceFS._fail_if_archived_or_sparse_file (/root/node_modules/noobaa-core/src/sdk/namespace_fs.js:1280:19)
    at async NamespaceFS.read_object_stream (/root/node_modules/noobaa-core/src/sdk/namespace_fs.js:477:13)
(node:9) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To termin
ate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 7)
```

I think it's an issue with sparse file - a dup for #6680

**Comment by nimrod-becker** [comment]
Should be fixed with #6698

**Comment by davidbw246** [comment]
I tested on noobaa 0816 - this issue is fixed.
